src/java/org/apache/cassandra/service/reads/DataResolver.java

```diff
@@ -121,11 +121,15 @@ private boolean needsReplicaFilteringProtection()
         if (command.rowFilter().isEmpty())
             return false;
 
-        IndexMetadata indexDef = command.indexMetadata();
-        if (indexDef != null && indexDef.isCustom())
+        ColumnFamilyStore cfs = ColumnFamilyStore.getIfExists(command.metadata().keyspace, command.metadata().name);
```

Comment:
I think that for better performance we should check whether the `IndexMetadata` is null or not-custom before opening `ColumnFamilyStore`. Also, we can open the store with the shorter `ColumnFamilyStore.getIfExists(command.metadata().id)`.